packages/embark-ui/src/containers/TransactionsContainer.js

```diff
@@ -34,34 +33,56 @@ class TransactionsContainer extends Component {
   }
 
   get numberOfBlocks() {
-    const blocks = this.props.blocks;
-    return !blocks.length ? 0 : blocks[0].number + 1;
+    if (this._numberOfBlocks === undefined) {
+      const blocks = this.props.blocks;
+      this._numberOfBlocks = !blocks.length ? 0 : blocks[0].number + 1;
+    }
+    return this._numberOfBlocks;
   }
 
   get estNumberOfTxs() {
-    const blocks = this.props.blocks;
-    const numBlocksInProps = blocks.length;
-    const numTxsInPropsBlocks = blocks.reduce((txCount, block) => (
-      txCount + block.transactions.length
-    ), 0);
-    const missingNumBlocks = this.numberOfBlocks - numBlocksInProps;
-    return missingNumBlocks + numTxsInPropsBlocks;
+    if (this._estNumberOfTxs === undefined) {
```

Comment:
```suggestion
    if (this._estNumberOfTxs !== undefined) {
      return this._estNumberOfTxs;
   }
```
Just to have less code in the if

Reply:
Trade-off is that another `return` statement is needed since either way it needs to `return this._estNumberOfTxs`.

Reply:
Done.